lib/game/game.dart

```diff
@@ -101,30 +107,52 @@ abstract class Game {
 
   /// Use this method to load the assets need for the game instance to run
   Future<void> onLoad() async {}
-
-  /// Returns the widget which will be show while the instance is loading
-  Widget loadingWidget() => Container();
 }
 
-class OverlayWidget {
-  final String name;
-  final Widget widget;
+/// A [ChangeNotifier] used to control the visibility of overlays on a [Game].
+///
+/// To learn more, see:
+/// - [HasWidgetsOverlay]
+/// - [GameWidget.withOverlay]
+class ActiveOverlaysNotifier extends ChangeNotifier {
+  final Set<String> _activeOverlays = {};
+
+  bool add(String overlayName) {
+    final setChanged = _activeOverlays.add(overlayName);
+    if (setChanged) {
+      notifyListeners();
+    }
+    return setChanged;
+  }
+
+  bool remove(String overlayName) {
+    final hasRemoved = _activeOverlays.remove(overlayName);
+    if (hasRemoved) {
+      notifyListeners();
+    }
+    return hasRemoved;
+  }
 
-  OverlayWidget(this.name, this.widget);
+  Set<String> get value => _activeOverlays;
 }
 
+/// A mixin on [Game] to use the widget overlay API.
+///
+/// Mix a game with it and use alongside [GameWidget.withOverlay] to have overlays.
+///
+/// This is useful to render widgets above your game, like a pause menu for example.
+/// After mixing this in a [Game] subclass, use the [overlays] property to mark
+/// overlays visible or hidden via [overlays.add] or [overlays.remove], respectively.
+///
+/// Ex:
+/// ```
+/// final pauseOverlayIdentifier = "PauseMenu";
+/// overlays.add(pauseOverlayIdentifier); // marks "PauseMenu" to be renderable.
+/// overlays.remove(pauseOverlayIdentifier); // marks "PauseMenu" to not be renderable.
+/// ```
+///
+/// To learn more, see:
+/// - [GameWidget.withOverlay]
 mixin HasWidgetsOverlay on Game {
```

Comment:
Since this is always available on the gamewidget, I would take a leap here and say that we don't need this anymore, maybe we could just make this a default feature on the game class.